bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -606,16 +607,26 @@ void scheduleBookieHealthCheckIfEnabled(ClientConfiguration conf) {
             scheduler.scheduleAtFixedRate(new SafeRunnable() {
 
                 @Override
-                public void safeRun() {
+                public void safeRun() throws ExecutionException, InterruptedException {
                     checkForFaultyBookies();
                 }
                     }, conf.getBookieHealthCheckIntervalSeconds(), conf.getBookieHealthCheckIntervalSeconds(),
                     TimeUnit.SECONDS);
         }
     }
 
-    void checkForFaultyBookies() {
+    void checkForFaultyBookies() throws ExecutionException, InterruptedException {
         List<BookieId> faultyBookies = bookieClient.getFaultyBookies();
+        if (faultyBookies.isEmpty()) {
+            return;
+        }
+
+        boolean isEnable = metadataDriver.isHealthCheckEnabled().get();
```

Comment:
you can catch "ExecutionException, InterruptedException" here and do not add the throws clause to checkForFaultyBookies()

Reply:
like this?
        boolean isEnable = false;
        try {
            isEnable = metadataDriver.isHealthCheckEnabled().get();
        } catch (InterruptedException e) {
            LOG.error("isHealthCheckEnabled throw InterruptedException", e);
        } catch (ExecutionException e) {
            LOG.error("isHealthCheckEnabled throw ExecutionException", e);
        }
        if (!isEnable) {
            LOG.info("Health checks is currently disabled!");
            return;
        }